### PR TITLE
CT-1307 Allow users to remove links between cases

### PR DESCRIPTION
**Pull request by aliuk2012**
- the "Remove link" on the case detail page
- the remove_linked_case method to Case::Base model
- The controller action to remove the link.
- add destroy method to case linking service
- add new event to the state machine.
- trigger the event from CaseLinkingService#destroy

### CODE REVIEW

File: app/services/case_linking_service.rb [hunk 1]
```diff
@@ -34,6 +34,34 @@ def call
     @result = :error
   end
 
+  def destroy
+    ActiveRecord::Base.transaction do
+      # find the linked case
+      @link_case = Case::Base.where(number: @link_case_number).first
```
Comment (misaka) [inline app/services/case_linking_service.rb]: I think `@case.linked_cases.find_by(number: @link_case_number).first` would make more sense and be safer here.

File: spec/controllers/cases_controller/show_spec.rb [hunk 2]
```diff
@@ -194,7 +195,7 @@
         let(:user) { another_responder }
 
         it 'permitted_events to be empty' do
```
Comment (misaka) [inline spec/controllers/cases_controller/show_spec.rb]: Can we change the spec description to match the actual expectation?

File: spec/controllers/cases_controller/show_spec.rb [hunk 3]
```diff
@@ -260,7 +261,7 @@
         let(:user) { create(:responder) }
 
         it 'permitted_events to be empty' do
```
Comment (misaka) [inline spec/controllers/cases_controller/show_spec.rb]: Can we change the spec description to match the actual expectation?

File: spec/controllers/cases_controller/show_spec.rb [hunk 4]
```diff
@@ -323,7 +324,7 @@
...
```
Comment (misaka) [inline spec/controllers/cases_controller/show_spec.rb]: Can we change the spec description to match the actual expectation?

File: spec/controllers/cases_controller/show_spec.rb [hunk 5]
```diff
@@ -371,7 +372,7 @@
         let(:user) { responder }
 
         it 'permitted_events to be empty' do
```
Comment (misaka) [inline spec/controllers/cases_controller/show_spec.rb]: Can we change the spec description to match the actual expectation?

File: spec/controllers/cases_controller/show_spec.rb [hunk 6]
```diff
@@ -384,7 +385,7 @@
         let(:user) { create(:responder) }
 
         it 'permitted_events to be empty' do
```
Comment (misaka) [inline spec/controllers/cases_controller/show_spec.rb]: Can we change the spec description to match the actual expectation?

File: spec/views/cases/_linked_cases_html_slim_spec.rb [hunk 7]
```diff
@@ -51,6 +56,12 @@ def disallow_case_policy(policy_name, kase)
         expect(row.case_type.text).to eq 'FOI '
         expect(row.request.text)
             .to eq "#{ linked_case.subject } #{ linked_case.name }"
+
+        # annoyingly had to add \n below because of how locales using html is handled
```
Comment (misaka) [inline spec/views/cases/_linked_cases_html_slim_spec.rb]: Just to mention, you could also do `row.remove_link.text.strip` to remove whitespace.

**Review by misaka — CHANGES_REQUESTED** [review]

File: app/services/case_linking_service.rb (hunk 1 above)
Comment (aliuk2012) [reply app/services/case_linking_service.rb]: 👍

**Comment by misaka** [comment]
As discussed with @aliuk2012, the changes to the spec descriptions are actually highlighting a bigger issue ... that we need to refactor how we present action buttons and links in general. We'll save this for another ticket.

**Review by misaka — APPROVED** [review]

**Pull request merged**